provider/kubernetes/kubernetes.go

```diff
@@ -137,16 +139,56 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 					return nil
 				}
 			}
+
+			eventsChanToRead := eventsChan
+			throttleDuration := time.Duration(p.ThrottleDuration)
+			if throttleDuration > 0 {
```

Comment:
I think you could have all of 145-174 in a

func throttleEvents(throttleDuration time.Duration) chan interface{} {
  if throttleDuration <= 0 {
    return nil
  }
  ....
  return eventsChanBuffered
}

then at 145 you could do:

throttledChan := throttleEvents()
if throttledChan != nil {
  eventsChanToRead = throttledChan
}

for less indented code and better readability.